### DISCUSSION TITLE
SOFA UNITY ERRORS

### DISCUSSION THREAD

**Discussion by a58759578** [InfinyTech3D/SofaUnity#78] [Help - Using SofaUnity]
Hi, I'm having some problems running the sample scenes,the unity3d always crashes when running these cases,please help me take a look:
My Unity3d Version:2021.3.21f1c1
1.Scenes/Demos/Haptic/Geomagic/:    All examples of this function report errors and cannot run properly.The errors are as follows:
logs:
```
SofaContextAPI::loadPlugin: G:/Projects/SOFA/Sofa_Unity_Project/Assets/SofaUnity/Plugins/Native/x64//Geomagic.dll, method returns: -12: Sofa plugin loading failed. Cause: invalid loading.
UnityEngine.Debug:LogError (object)
SofaUnityAPI.SofaContextAPI:loadPlugin (string) (at Assets/SofaUnity/Plugins/SofaUnityAPI/SofaContextAPI.cs:240)
SofaUnity.PluginManagerInterface:LoadPlugins () (at Assets/SofaUnity/Scripts/Core/System/PluginManager.cs:254)

ExecutionEngineException: String conversion error: Illegal byte sequence encounted in the input.
SofaUnityAPI.SofaContextAPI.DisplayMessages () (at Assets/SofaUnity/Plugins/SofaUnityAPI/SofaContextAPI.cs:474)
SofaUnity.SofaContext.DoCatchSofaMessages () (at Assets/SofaUnity/Scripts/SofaContext.cs:565)
```
![image](https://github.com/InfinyTech3D/SofaUnity/assets/34824384/a7c96a5a-4fb6-4658-9292-f7311fd2a326)

2.Scenes/Demos/Imaging/Ultrasound/:
logs:
```
SofaContextAPI::loadPlugin: G:/Projects/SOFA/Sofa_Unity_Project/Assets/SofaUnity/Plugins/Native/x64//ImagingUS.dll, method returns: -12: Sofa plugin loading failed. Cause: invalid loading.
UnityEngine.Debug:LogError (object)
SofaUnityAPI.SofaContextAPI:loadPlugin (string) (at Assets/SofaUnity/Plugins/SofaUnityAPI/SofaContextAPI.cs:240)

ExecutionEngineException: String conversion error: Illegal byte sequence encounted in the input.
SofaUnityAPI.SofaContextAPI.DisplayMessages () (at Assets/SofaUnity/Plugins/SofaUnityAPI/SofaContextAPI.cs:474)
SofaUnity.SofaContext.DoCatchSofaMessages () (at Assets/SofaUnity/Scripts/SofaContext.cs:565)
```
![image (1)](https://github.com/InfinyTech3D/SofaUnity/assets/34824384/a99e02ae-b085-451b-8bae-f40c2d1c4779)

**Comment by epernod** [comment]
Hi,

Could you give the version of the asset you are using? v22.12 or v23.06?
Could you also confirm that the classic examples scenes are working? Like example_01_Liver ?

Best regards

**Comment by epernod** [comment]
Hi,

as I didn't get any response from you, I'm closing this discussion thread. Feel free to reopen it or ask other questions in another thread.

Best regards,
